spec/models/hmcts_common_platform/defendant_summary_spec.rb

```diff
@@ -52,6 +54,47 @@
       expect(json["proceedings_concluded"]).to be false
       expect(json["representation_order"]).to be_present
       expect(json["offence_summaries"].count).to be(2)
+      expect(json["application_summaries"].count).to be(1)
+    end
+  end
+
+  describe "#match_application_summaries" do
+    let(:matching_court_application) do
+      {
+        "applicationId" => "f38d0030-0b4a-4fa5-9484-bb37b1e6ab39",
+        "subjectSummary" => {
+          "masterDefendantId" => "DEF123",
+        },
+      }
+    end
+
+    let(:not_matching_court_application) do
+      {
+        "applicationId" => "f38d0030-0b4a-4fa5-9484-bb37b1e6ac56",
+        "subjectSummary" => {
+          "masterDefendantId" => "DEF145",
+        },
+      }
+    end
+
+    let(:defendant_summary) do
+      described_class.new(
+        { defendantId: "DEF123" },
+        [matching_court_application, not_matching_court_application],
+      )
+    end
+
+    it "includes the summary when the masterDefendantId matches the defendantId" do
+      summaries = defendant_summary.application_summaries
+
+      expect(summaries).not_to be_empty
+      expect(summaries.first).to be_a(HmctsCommonPlatform::ApplicationSummary)
+      expect(summaries.first.data["applicationId"]).to eq("f38d0030-0b4a-4fa5-9484-bb37b1e6ab39")
+    end
+
+    it "does not include the summary when masterDefendantId does not match defendantId" do
+      summaries = defendant_summary.application_summaries
+      expect(summaries.first.data["applicationId"]).not_to eq("f38d0030-0b4a-4fa5-9484-bb37b1e6ac56")
```

Comment:
The problem with this test is that if the code didn't work, the test would still pass. You've passed in `...ab39` and `...ac56` to the defendant_summary. If filtering wasn't working, then `defendant_summaries.application_summaries` would contain both `...ab39` and `...ac56`. This test only checks that the _first_ application summary isn't `...ac56`... but if the code was buggy, the first application summary would still almost certainly be `...ab39`, so this would pass!